src/config/environmentConfig.ts

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+/**
+ * Validates and sets the environment configuration based on the provided schema.
+ * If the environment config is invalid, an error is thrown.
+ *
+ * @throws {Error} If the environment config is invalid
+ *
+ * @returns {EnvironmentConfig} The validated environment configuration
+ */
+
+import * as joi from '@hapi/joi';
+import { EnvironmentConfig } from '../models/environmentConfig.interface';
```

Comment:
interfaces can go into a different folder like interfaces, but i am not convinced that is a part of models.

Reply:
The idea was not to put all the interfaces, schemas, and type definitions inside the `src/models` folder with their respective folders inside it. Just want to avoid having too many folders inside `src`. If the `models` don't look appropriate, can you suggest something else or you prefer having folder inside `src`?

Now this has been moved to `src/interfaces`